<?xml version="1.0"?>
<!-- This is the launch file used to run the robot, specifically Baxter.-->
  <launch>
    <!-- These lines are used to define the initializeable arguments and parameters-->
    <!-- Argument "pedal" is set to "1" when pedal option is used.-->
    <arg name="pedal" default="0"/>
    <arg name="jump" default="0"/>
    <arg name="number" default="1"/>
    <arg name="control" default="1"/>
    <!-- Assigning the argument "pedal" value to parameter "use_pedal" -->
    <param name="use_pedal" value="$(arg pedal)" type="int"/>
    <param name="jump_mode" value="$(arg jump)" type="int"/>
    <param name="number_screen" value="$(arg number)" type="int"/>
    <param name="baxter_control" value="$(arg control)" type="int"/>
    <!-- This node is run when the pedal is not used -->
    <group unless="$(arg pedal)">
      <!-- Used to receive the target position and move the Baxter toward the target-->
       <node pkg="kcl_ergonomics" type="pose_orientation_control.py" name="pose_orientation_control" respawn="false" output="screen"/>
    </group>

    <!-- This node is run when the pedal is used -->
    <group if="$(arg pedal)">
      <!-- Used to receive the target position and move the Baxter toward the target-->
       <node pkg="kcl_ergonomics" type="pose_orientation_control_pedal.py" name="pose_orientation_control" respawn="false" output="screen"/>
    </group>

    <!-- This node is used to create a TF listener and publish euler angle of tip's orientation -->
    <!-- CHECK THIS! IT MIGHT NOT BE NEEDED! -->
    <node pkg="kcl_ergonomics" type="listen_orientation.py" name="listen_orientation" respawn="false" output="screen"/>

    <!-- Used to show Rula score image in Baxter's screen-->
    <node pkg="kcl_ergonomics" type="image_screen.py" name="image_screen" respawn="false" output="screen"/>
  </launch>
